File: 2523. Closest Prime Numbers in Range/Solution.py
class Solution:
    def closestPrimes(self, left: int, right: int) -> List[int]:
        sieve = [True for i in range(right + 1)]
        resFirst, resSecond, prev, diff = -1, -1, -1, 1000000
        for i in range(2, len(sieve)):
            if not sieve[i]: continue
            for j in range(i + i, right + 1, i): sieve[j] = False
            if i >= left:
                if prev != -1 and i - prev < diff: resFirst, resSecond, diff = prev, i, i - prev
                prev = i
        return [resFirst, resSecond]

File: 2523. Closest Prime Numbers in Range/Solution.java
class Solution {
    public int[] closestPrimes(int left, int right) {
        boolean[] sieve = new boolean[right + 1];
        int resFirst = -1, resSecond = -1, prev = -1, diff = 1000000;
        for (int i = 2; i < right + 1; i++) {
            if (sieve[i]) continue;
            for (int j = i + i; j < right + 1; j += i) sieve[j] = true;
            if (i >= left) {
                if (prev != -1 && i - prev < diff) {
                    resFirst = prev;
                    resSecond = i;
                    diff = i - prev;
                }
                prev = i;
            }
        }
        return new int[] { resFirst, resSecond };
    }
}
